specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/webapplicationfirewall.json

```diff
@@ -0,0 +1,575 @@
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "DeleteWebApplicationFirewallPolicy"
+        ],
+        "description": "Deletes Policy",
+        "operationId": "Policies_Delete",
+        "x-ms-examples": {
+          "Deletes a WAF policy within a resource group": {
+            "$ref": "./examples/WafPolicyDelete.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "policyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 128,
+            "description": "The name of the policy."
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted. The request has been accepted for processing and the operation will complete asynchronously."
+          },
+          "204": {
+            "description": "No Content. The request has been accepted but the policy was not found."
+          },
+          "200": {
+            "description": "Delete successful."
+          }
+        },        
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "WebApplicationFirewallPolicy": {
+      "description": "Defines web application firewall policy.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/WebApplicationFirewallPolicyPropertiesFormat",
+          "description": "Properties of the web application firewall policy."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+            "$ref": "./network.json#/definitions/Resource"
+        }
+      ]
+    },
+    "WebApplicationFirewallPolicyPropertiesFormat": {
+      "description": "Defines web application firewall policy properties",
+      "properties": {
+        "policySettings": {
+          "description": "Describes  policySettings for policy",
+          "$ref": "#/definitions/policySettings"
+        },
+        "customRules": {
+          "description": "Describes custom rules inside the policy",
+          "$ref": "#/definitions/CustomRules"
+        },
+        "applicationGateways": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./applicationGateway.json#/definitions/ApplicationGateway"
```

Comment:
If this array contains only id-objects (`{ "id": "..." }) I think using `./network.json#/definitions/SubResource` would be better
@jianghaolu please confirm

Reply:
@number213 not in the case of network. Their models are read-write so on a GET all the other properties will be populated.

This is technically a security violation because users with READ access to a Firewall may not have access to AppGateway. But it's been the way it is 😱